src/Array/Extra.elm

```diff
@@ -312,3 +313,30 @@ removeAt index xs =
 
     else
         append xs0 (slice 1 len1 xs1)
+
+
+{-| Insert an element at the given index.
+
+    insertAt 1 'b' (fromList [ 'a', 'c' ]) == fromList [ 'a', 'b', 'c' ]
+
+    insertAt -1 'b' (fromList [ 'a', 'c' ]) == fromList [ 'b', 'a', 'c' ]
+
+    insertAt 10 'b' (fromList [ 'a', 'c' ]) == fromList [ 'a', 'c', 'b' ]
+
+-}
+insertAt : Int -> a -> Array a -> Array a
+insertAt index val values =
+    let
+        length =
+            Array.length values
+
+        index_ =
+            clamp 0 length index
```

Comment:
What is `clamp`? It doesn't seem to be in  scope in this file.

Reply:
https://package.elm-lang.org/packages/elm/core/latest/Basics#clamp